p2panda-store/src/lib.rs

```diff
@@ -1,5 +1,19 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 
+//! Interfaces and implementations of persistence layers for core p2panda data types.
+//!
+//! The provided APIs allow for efficient implementations of `Operation` and `Log` stores. These
+//! persistence and query APIs are utilised by higher-level components of the `p2panda` stack, such
+//! as `p2panda-sync` and `p2panda-stream`. For detailed information concerning the `Operation` and
+//! `Log` types, please consult the documentation for the `p2panda-core` crate.
+//!
+//! The traits provided here are not intended to offer generic storage solutions for non-`p2panda`
+//! data types, nor are they intended to solve application-layer storage concerns.
+//!
+//! An in-memory storage solution is provided in the form of a `MemoryStore` which implements both
+//! `OperationStore` and `LogStore`. The store is gated by the `memory` feature flag and is enabled
+//! by default.
+
```

Comment:
Remove new-line here (we don't do them for both lib-level or mod-level doc-strings)